### PR TITLE
Test coff arch refactor

### PR DESCRIPTION
**Pull request by galenbwill**
COFF loading support complete

### CODE REVIEW

File: arch_armv7.cpp [hunk 1]
```diff
@@ -2327,20 +2419,536 @@ class ArmMachORelocationHandler: public RelocationHandler
...
+		if (len < info.size)
+		{
+			return false;
+		}
+
+		//auto swap = [&arch](uint32_t x) { return (arch->GetEndianness() == LittleEndian)? x : bswap32(x); };
+		switch (info.nativeType)
+		{
+		case PE_IMAGE_REL_THUMB_MOV32:
+			if (true || arch->GetName() == "thumb2")
```
Comment (plafosse) [inline arch_armv7.cpp]: ? what's this condition for?

File: arch_armv7.cpp [hunk 2]
```diff
@@ -2327,20 +2419,536 @@ class ArmMachORelocationHandler: public RelocationHandler
...
+				_mov* movt = (_mov*)dest32 + 1;
+				if (movt->is_movt_flag != 1) // movt->group2_4 != MOVT)
+				{
+					LogWarn("Expected MOVT in 0x%08" PRIx32 " (0x%" PRIx16 ") at 0x%" PRIx64 " but found 0x%" PRIx32 " (0x%" PRIx32 ") movt_flag: %d",
+						movt->word, MOVT, address + 4, movt->group2_4, *(dest32 + 1), movt->is_movt_flag);
+				}
+
+				_target *targetHiLo = (_target*)&target;
+
+				// This could be done more efficiently with shifts, ands, and ors, but that's the compiler's job
```
Comment (plafosse) [inline arch_armv7.cpp]: 🤌

**Review by plafosse — CHANGES_REQUESTED** [review]
that one line I made a comment on probably needs to change, other then that it looks good

File: arch_armv7.cpp (hunk 1 above)
Comment (galenbwill) [reply arch_armv7.cpp]: an unnecessary check that should have been removed. nice catch!

**Comment by galenbwill** [comment]
this PR is out of date

**Pull request closed without merging**